imap_processing/ialirt/l0/process_swapi.py

```diff
@@ -3,17 +3,122 @@
+    """
+    Find the pseudo speed (u), density (n) and temperature (T) of solar wind.
+
+    Fit a curve to calculated count rate values as a function of energy passbands.
+
+    Parameters
+    ----------
+    count_rates : np.ndarray
+        Particle coincidence count rates.
+
+    Returns
+    -------
+    solution_dict : dict
+        Dictionary containing the optimized speed, density, and temperature values for
+        each sweep included in the input count_rates array.
+    """
+    # Read in energy passbands
+    energy_data = pd.read_csv(
+        f"{imap_module_directory}/tests/ialirt/test_data/ialirt_test_data.csv"
+    )
+    energy_passbands = energy_data["Energy [eV/q]"].to_numpy()
+
+    def count_rate(
+        energy_pass: float, speed: float, density: float, temp: float
+    ) -> float | np.ndarray:
+        """
+        Compute SWAPI count rate for provided E_e, u, n, T.
+
+        Parameters
+        ----------
+        energy_pass : float
+            Energy passband [eV].
+        speed : float
+            Bulk solar wind speed [km/s].
+        density : float
+            Proton density [cm^-3].
+        temp : float
+            Temperature [K].
+
+        Returns
+        -------
+        count_rate : float | np.ndarray
+            Particle coincidence count rate.
+        """
+        # Scientific constants used in optimization model
+        boltz = 1.380649e-23  # Boltzmann constant, J/K
+        at_mass = 1.6605390666e-27  # atomic mass, kg
+        prot_mass = 1.007276466621 * at_mass  # mass of proton, kg
+        eff_area = 3.3e-5 * 1e-4  # effective area, meters squared
+        az_fov = np.deg2rad(30)  # azimuthal width of the field of view, radians
+        fwhm_width = 0.085  # FWHM of energy width
+        speed_energy_width = 0.5 * fwhm_width  # speed width of energy passband
+
+        # thermal velocity of solar wind ions
+        thermal_velocity = np.sqrt(2 * boltz * temp / prot_mass)
+        beta = 1 / (thermal_velocity**2)
+        # convert energy to Joules
+        center_speed = np.sqrt(2 * energy_pass * 1.60218e-19 / prot_mass)
+        speed = speed * 1000  # convert km/s to m/s
+        density = density * 1e6  # convert 1/cm**3 -to 1/m**3
+
+        return (
+            (density * eff_area * (beta / np.pi) ** (3 / 2))
+            * (np.exp(-beta * (center_speed**2 + speed**2 - 2 * center_speed * speed)))
+            * np.sqrt(np.pi / (beta * speed * center_speed))
+            * erf(np.sqrt(beta * speed * center_speed) * (az_fov / 2))
+            * (
+                center_speed**4
+                * speed_energy_width
+                * np.arcsin(thermal_velocity / center_speed)
+            )
+        )
+
+    initial_param_guess = np.array([550, 5.27, 1e5])
+    solution_dict = {  # type: ignore
+        "pseudo_speed": [],
+        "pseudo_density": [],
+        "pseudo_temperature": [],
+    }
+
+    if count_rates.ndim > 1:
```

Comment:
Are we expecting both cases of array and float inputs? Can you just do one or the other and fix what the input is here.